### PR TITLE
Feature/unity analytics

### PR DESCRIPTION
**Pull request by baconeta**
This PR adds base functionality for Unity Analytics to work. 
For it to work out of the box, you will first need to set the project settings with correct values given from Unity Game Services setup. Otherwise, analytics will fail to authenticate and send. This is detailed in the updated readme file. 

I have also included some template code and structure for the analytics events.

Note: the editor should prompt for Unity Gaming Services linking on first open.

### CODE REVIEW

File: Assets/Scripts/Analytics/TimingEventAnalytics.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    /// Use a new instance of this script for each event to be timed.
+    /// </summary>
+    public class TimingEventAnalytics : MonoBehaviour
+    {
+        [SerializeField] private string analyticStartedName;
+        [SerializeField] private string analyticEndedName;
+        [SerializeField] private string analyticEndTimeName;
+        [SerializeField] private bool sendStartEvent = true;
+
+        private float _cachedTimeCreditsOpen;
```
Comment (TomasMorton) [inline Assets/Scripts/Analytics/TimingEventAnalytics.cs]: There's a few references to 'credits' in this file, but the actual code seems quite generic, is that intentional?

**Comment by TomasMorton** [comment]
Success (except that I didn't define a custom event in the Event Manager):
image

Steps I took:
- Create a unity project
- Open Unity (it installed the package necessary)
- Clicked the button to link the project when I was prompted
- Added the UnityAnalytics script to a new game object
- Added the TimingEventAnalytics script to the same game object
- Create two button, each with an OnClick to call start/end timer.
- Viewed results in dashboard.

I'd note that because analytics are slow, I added a Debug.log in the TimingEventAnalytics  methods so that I could tell it was working.

**Review by TomasMorton — APPROVED** [review]

File: Assets/Scripts/Analytics/TimingEventAnalytics.cs (hunk 1 above)
Comment (baconeta) [reply Assets/Scripts/Analytics/TimingEventAnalytics.cs]: It is not - good catch :D

**Comment by baconeta** [comment]
> Success (except that I didn't define a custom event in the Event Manager): image
> 
> Steps I took:
> 
> * Create a unity project
> * Open Unity (it installed the package necessary)
> * Clicked the button to link the project when I was prompted
> * Added the UnityAnalytics script to a new game object
> * Added the TimingEventAnalytics script to the same game object
> * Create two button, each with an OnClick to call start/end timer.
> * Viewed results in dashboard.
> 
> I'd note that because analytics are slow, I added a Debug.log in the TimingEventAnalytics methods so that I could tell it was working.

Note that you can enable the scripting debug mode for analytics by doing:
1.  In the Unity Editor, go to Edit > Project Settings > Player Settings > Player.
2.  Locate the Script Compilation section and select Scripting Define Symbols.
3.  Add the keyword “UNITY_ANALYTICS_EVENT_LOGS”.
4.  Select Apply.

**Review by TomasMorton — APPROVED** [review]

**Pull request merged**